src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -0,0 +1,247 @@
+		return jsonArray;
+	}
+
+	/**
+	 * Generates the documentation JsonObject for an element that is annotated with documentation
+	 * annotations (e.g. effects, conditions, etc.)
+	 * @param syntaxInfo the syntax info element to generate the documentation object of
+	 * @return the JsonObject representing the documentation of the provided syntax element
+	 */
+	private @Nullable JsonObject generatedAnnotatedElement(SyntaxElementInfo<?> syntaxInfo) {
+		Class<?> syntaxClass = syntaxInfo.getElementClass();
+		Name nameAnnotation = syntaxClass.getAnnotation(Name.class);
+		if (nameAnnotation == null || syntaxClass.getAnnotation(NoDoc.class) != null)
+			return null;
+		JsonObject syntaxJsonObject = new JsonObject();
+		syntaxJsonObject.addProperty("id", DocumentationIdProvider.getId(syntaxInfo));
+		syntaxJsonObject.addProperty("name", nameAnnotation.value());
+
+		Since sinceAnnotation = syntaxClass.getAnnotation(Since.class);
+		syntaxJsonObject.addProperty("since", sinceAnnotation.value());
+
+		Description descriptionAnnotation = syntaxClass.getAnnotation(Description.class);
+		if (descriptionAnnotation != null)
+			syntaxJsonObject.add("description", convertToJsonArray(descriptionAnnotation.value()));
+		else
+			syntaxJsonObject.add("description", new JsonArray());
+
+		Examples examplesAnnotation = syntaxClass.getAnnotation(Examples.class);
+		if (examplesAnnotation != null) {
+			syntaxJsonObject.add("examples", convertToJsonArray(examplesAnnotation.value()));
+		} else {
+			syntaxJsonObject.add("examples", new JsonArray());
+		}
+
+
+		syntaxJsonObject.add("patterns", convertToJsonArray(syntaxInfo.getPatterns()));
+		return syntaxJsonObject;
+	}
+
+	/**
+	 * Generates the documentation JsonObject for an event
+	 * @param eventInfo the event to generate the documentation object for
+	 * @return a documentation JsonObject for the event
+	 */
+	private JsonObject generateEventElement(SkriptEventInfo<?> eventInfo) {
+		JsonObject syntaxJsonObject = new JsonObject();
+		syntaxJsonObject.addProperty("id", DocumentationIdProvider.getId(eventInfo));
+		syntaxJsonObject.addProperty("name", eventInfo.name);
+		syntaxJsonObject.addProperty("since", eventInfo.getSince());
+		syntaxJsonObject.add("description", convertToJsonArray(eventInfo.getDescription()));
+		syntaxJsonObject.add("examples", convertToJsonArray(eventInfo.getExamples()));
+		syntaxJsonObject.add("patterns", convertToJsonArray(eventInfo.patterns));
+		return syntaxJsonObject;
+	}
+
+
+	/**
+	 * Generates a JsonArray containing the documentation JsonObjects for each structure in the iterator
+	 * @param infos the structures to generate documentation for
+	 * @return a JsonArray containing the documentation JsonObjects for each structure
+	 */
+	private <T extends StructureInfo<? extends Structure>> JsonArray generateStructureElementArray(Iterator<T> infos) {
+		JsonArray syntaxArray = new JsonArray();
+		infos.forEachRemaining(info -> {
+			if (info instanceof SkriptEventInfo) {
```

Comment:
```suggestion
			if (info instanceof SkriptEventInfo eventInfo) {
```
if possible with generics